modules/tensor_mechanics/src/materials/ComputeEigenstrain.C

```diff
@@ -0,0 +1,33 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeEigenstrain.h"
+
+template<>
+InputParameters validParams<ComputeEigenstrain>()
+{
+  InputParameters params = validParams<ComputeStressFreeStrainBase>();
+  params.addClassDescription("Computes a constant Eigenstrain");
+  params.addRequiredParam<std::vector<Real> >("eigen_base","Vector of values defining the constant base tensor for the Eigenstrain");
+  params.addParam<std::string>("prefactor", "var_dep", "Name of material defining the variable dependence");
```

Comment:
Having a required material property here, even if the user doesn't want a prefactor, is problematic.